### DISCUSSION TITLE
Handling the database

### DISCUSSION THREAD

**Discussion by jacob-carlborg-apoex** [shakacode/cypress-playwright-on-rails#126] [Q&A] [answered]
I've just started to try Cypress and this gem and I'm wondering how to handle the database. There are two things I'm thinking of:

1. According to the best Cypress practice [1] they recommend against cleaning up after tests have been run and instead clean up **before** each test is run. The intention is to leave the sate produced by the test intact. Therefore if you follow this best practice and first run Cypress tests and the regular Rails test they can fail because they expect an empty database. How are you handling this? I can think of two solutions: having two separate databases for the Cypress tests and for the other tests. The other alternative is to clean the database in a before suite hook for the regular Rails tests.

1. How to handle migrations? One idea I had was to use the `config.active_record.maintain_test_schema` [2] Rails configuration option. Perhaps set it to `true` only if the `CYPRESS` environment variable is set to true.

Thoughts, ideas?

[1] https://docs.cypress.io/guides/references/best-practices#Using-after-Or-afterEach-Hooks
[2] https://guides.rubyonrails.org/#config-active-record-maintain-test-schema

**Comment by justin808** [comment, marked as answer]
Hi @jacob-carlborg-apoex, with cypress-on-rails, you can configure your test fixtures like rspec tests. Did you find that not to be the case?

With regards to migrations, again, this is no different than other rspec tests. Why would migrations be related to such tests?

**Reply by jacob-carlborg-apoex** [reply]
> with cypress-on-rails, you can configure your test fixtures like rspec tests. Did you find that not to be the case?

I'm not sure I understand, could you please elaborate? Since the regular Rails tests are run in a transaction it will always clean up after itself and therefore not need to clean the database before the test suite is run, or is it doing that anyway just to be sure?

> With regards to migrations, again, this is no different than other rspec tests. Why would migrations be related to such tests?

Before now I have never thought of how the test database is kept in sync with the migrations. I noticed in our application there's a called to `ActiveRecord::Migration.maintain_test_schema!` in `spec/rails_helper.rb` (we're using RSpec). If I run the Cypress tests that file shouldn't be loaded?

**Comment by Judahmeek** [comment]
@KhaledEmaraDev 
@grantspeelman 
Can either of you follow up on this?
